core/src/main/scala/io/finch/EndpointResult.scala

```diff
@@ -40,32 +40,29 @@ sealed abstract class EndpointResult[F[_], +A] {
     case _                                 => None
   }
 
-  def awaitOutput(d: Duration = Duration.Inf)(implicit F: Effect[F]): Option[Either[Throwable, Output[A]]] = this match {
+  def awaitOutput(d: Duration = Duration.Inf)(implicit dispatcher: Dispatcher[F]): Option[Either[Throwable, Output[A]]] = this match {
```

Comment:
I think we should ask for Async[F] and summon Dispatcher instead.

Something along the lines:

```
Dispatcher[F].use { dispatcher =>
case EndpointResult.Matched(_, _, out) =>
      try Some(Right(dispatcher.unsafeRunTimed(out, d)))
      catch {
        case _: TimeoutException => Some(Left(new TimeoutException(s"Output wasn't returned in time: $d")))
        case e: Throwable        => Some(Left(e))
      }
}
```

Reply:
Wouldn't that change the signature?